File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int n = nums.size(), l = -1, mid = -1, r = 0, maxLen = 0;

        while(r < n) {
            while(r < n && nums[r]) r++;
            maxLen = l > -1 ? max(maxLen, r - l - 2) 
                            : max(0, r - 1);
            l = mid;
            mid = r;
            r++;
        }
        return maxLen;
    }
};

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        def lengths_of_one_intervals():
            i, n = 0, len(nums)
            while i < n:
                j = i
                while i < n and nums[i] == 1:
                    i += 1
                yield i - j
                i += 1  # Skip one zero

        max_length = max((a + b for a, b in pairwise(lengths_of_one_intervals())), default=-1)
        return max_length if max_length != -1 else len(nums) - 1
        